samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/Apple.java

```diff
@@ -0,0 +1,128 @@
+package org.openapitools.model;
+
+import java.net.URI;
+import java.util.Objects;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.openapitools.jackson.nullable.JsonNullable;
+import java.time.OffsetDateTime;
+import javax.validation.Valid;
+import javax.validation.constraints.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+
+import java.util.*;
+import javax.annotation.Generated;
+
+/**
+ * Apple
+ */
+
+@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+public class Apple implements Fruit {
+
+  @JsonProperty("seeds")
+  private Integer seeds;
+
+  @JsonProperty("fruitType")
+  private FruitType fruitType;
+
+  /**
+   * Default constructor
+   * @deprecated Use {@link Apple#Apple(Integer)}
+   */
+  @Deprecated
+  public Apple() {
+    super();
+  }
+
+  /**
+   * Constructor with only required parameters
+   */
+  public Apple(Integer seeds) {
+    this.seeds = seeds;
+    this.fruitType = fruitType;
```

Comment:
Initialising `fruitType` like this is weird, but is a separate issue I think. If a class implements an interface which has a oneOf with a discriminator, the discriminator property is only half considered "required". The template adds initialisation of the field, but omits the constructor parameter. Arguably, the field should be initialised with the correct discriminator value, it should not be a constructor parameter at all.